cmake/modules/pre_dt.cmake

```diff
@@ -0,0 +1,90 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
> why should those functions not be provided as part of the dts CMake module ?

Take a look at this commit:

ef2c39b4a5 [[WIP] cmake: modules: add sysdts support](https://github.com/zephyrproject-rtos/zephyr/pull/52272/commits/ef2c39b4a5748330016adcfc38eed9f133c697b6)

In the RFC (https://github.com/zephyrproject-rtos/zephyr/pull/52272).

The pre_dt.cmake functionality is needed by both sysdts.cmake and dts.cmake. However, dts.cmake must run after sysdts.cmake in zephyr_default.cmake, because dts.cmake needs to already have the generated `.dts` created by sysdts.cmake when it starts to run.

To break the dependency cycle, I extracted the common functionality into a separate module.

If that's not satisfactory, do you have another suggestion for how to handle this?

Reply:
> To break the dependency cycle, I extracted the common functionality into a separate module.
> 
> If that's not satisfactory, do you have another suggestion for how to handle this?

Thanks for the link, and yes, I understand your reasoning.

atm I try to balance the continuous improvements, while keeping the longer term goal in mind.
#41301 took one step, by moving boilerplate code into CMake modules, but those modules are do a lot of stuff and relies a lot of external settings which is a bit unlike common CMake modules.
Common CMake module tends to provide functions which can be called, or information through CMake variables.

but taking such next step is outside scope of this PR, so will allow another day or two for extra considerations.